samples/HostedInAspNet.Server/Startup.cs

```diff
@@ -9,21 +10,27 @@ namespace HostedInAspNet.Server
 {
     public class Startup
     {
-        // This method gets called by the runtime. Use this method to add services to the container.
-        // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
         public void ConfigureServices(IServiceCollection services)
         {
+            services.AddBlazorCircuits();
+
+            services.AddHttpClient();
+            services.AddScoped<HttpClient>(s => s.GetRequiredService<IHttpClientFactory>().CreateClient());
```

Comment:
For my education, why do we need these lines to add `HttpClient`?

Reply:
ASP.NET Core doesn't add register an HttpClient as a service.

Reply:
Oh I see. So I guess we have a tradeoff between doing things in the more traditional server-centric manner (changing `FetchData.cshtml` to do `new HttpClient`) or retaining the Blazor WebAssembly-style model (getting `HttpClient` from the service provider).

Reply:
When we get Mono WebAssembly to work with `new HttpClient()` then I guess the solution becomes clear (stop using DI for that).

Reply:
Actually getting the client from the services is now the idiomatic thing to do in web app 👍

Reply:
I changed this to just use an service provided by the server side app. This is more idomatic for a server side app than making an http call to yourself, and demonstrates how these models differ.